core/controllers/editor.py

```diff
@@ -101,9 +101,53 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+class EditorLogoutHandler(base.BaseHandler):
+    """Handles logout from editor page."""
+
+    @acl_decorators.can_play_exploration
+    def is_valid_exploration_id(self, exp_id):
+        """This is to check if exploration id is valid."""
+        pass
+
+    @acl_decorators.open_access
+    def get(self):
+        """Checks if exploration is published and redirects accordingly."""
+
+        return_url = str(self.request.get('return_url'))
+        if return_url:
+            exploration_id = return_url.split('/')[-1]
+            self.is_valid_exploration_id(exploration_id)
```

Comment:
Drop this. You should check it in the call below perhaps.

Reply:
now i have used 'rights_manager.get_exploration_rights' to check if exp_id is valid.